### DISCUSSION TITLE
How to utilize Nvidia's Virtual Audio Device WDM

### DISCUSSION THREAD

**Discussion by joeblack2k2** [LizardByte/community#236] [Sunshine] [answered]
Hi to this great project! I'm running sunshine v19 stable on Windows 10 machine and try to switch from the GeforceExperience/Shield solution. Regarding the video I had no problems, but audio was since not satisfying. I'm running on an older PC with GTX1050 headlessly, for testing reasons just connecting hdmi-out to an dvi-monitor without sound transmission. Sunshine's audio-sink grabs the sound via default speakers (HD audio device) and streams to Moonlight client. Compared to the original GFE/shield audio connection the sunshine sound is way less quality (kind of reverb, somehow like analog phone quality).

So what I am trying to achieve ? I want use the same audio device, that Nvidias GFE/Shield also utilizes, which is based on NV Virtual Audio Device (WDM). Problem is: I cannot "catch" it - it is invisible in sound settings, no volume control etc (but the driver is correctly load in device manager). It gets only visible, when I start a streaming session via Original-Shield/GFE and disappears again, as soon as the session ends. Noting down the Device-ID during Shield-Session and passing that the Sunshine audio config as virtual_sink also doesnt work - error: cannot configure Wave, which seems to mean it cannot find the ID !?

Has anybody a hint, what to do to make the Nvidia Virtual Audio device visible/active ? Which technique do they use, to "hide" their Virtual audio device ?

Thanks in advance !

Joe

**Comment by cgutman** [comment, marked as answer]
Recent nightly builds (and v0.20.0 when it's released) use Steam's virtual audio device that accomplishes the same thing. Clear out the virtual sink and audio sink fields in your config and it will do it all automatically for you.

You can give it a try here: https://github.com/LizardByte/Sunshine/releases/tag/nightly-dev

**Reply by joeblack2k2** [reply]
Thanks for your reply. I gave it a try and probably will stick to it. The difference seems to be, that Steam acts like headphone (volume control) while Nvidia acts as speakers (constant volume, not controllable). The nightly is way better than the 19-stable :-) (Though has still a little more cracking in sound compared to the Nvidia).
